### PR TITLE
Refactor GatewayCleanup function

### PR DESCRIPTION
**Pull request by MaxBab**
The "GatewayCleanup" function restores the state of the GW node on the cluster if required.

The test should call the "SaveGatewayNodeForTest" function in order to save the state of the GW node into the Framework.

At the end of the test, the "GatewayCleanup" function will restore the state of the GW node if defined within the Framework state.

Signed-off-by: Maxim Babushkin <mbabushk@redhat.com>

### CODE REVIEW

File: test/e2e/framework/framework.go
Comment (tpantelis) [inline test/e2e/framework/framework.go]: This should be stored on the `Framework` instance and not globally. Also perhaps name it `gatewayNodesToReset`.

File: test/e2e/framework/gateways.go
Comment (tpantelis) [inline test/e2e/framework/gateways.go]: This will recreate the map on each call and erase any previous settings. The map should be created on construction of the `Framework` instance. Then this code becomes:

```suggestion
        f.gatewayNodesToReset[cluster] = append(f.gatewayNodesToReset[cluster], gwNode)
```

File: test/e2e/framework/gateways.go
Comment (tpantelis) [inline test/e2e/framework/gateways.go]: Replaced by the message below.

```suggestion
```

File: test/e2e/framework/gateways.go
Comment (tpantelis) [inline test/e2e/framework/gateways.go]:
```suggestion
                           By(fmt.Sprintf("Restoring gateway %q on cluster %q", gnode, TestContext.ClusterIDs[cluster]))
                           f.SetGatewayLabelOnNode(ClusterIndex(cluster), gnode, true)
```

File: test/e2e/framework/gateways.go
Comment (tpantelis) [inline test/e2e/framework/gateways.go]: This doesn't need to be public - it will be called internally by `DoFailover`. 

```suggestion
func (f *Framework) saveGatewayNode(cluster ClusterIndex, gwNode string) {
```

**Review by tpantelis — CHANGES_REQUESTED** [review]

File: test/e2e/framework/framework.go
Comment (MaxBab) [reply test/e2e/framework/framework.go]: Done.

File: test/e2e/framework/gateways.go
Comment (MaxBab) [reply test/e2e/framework/gateways.go]: Done

File: test/e2e/framework/gateways.go
Comment (MaxBab) [reply test/e2e/framework/gateways.go]: Done

File: test/e2e/framework/gateways.go
Comment (MaxBab) [reply test/e2e/framework/gateways.go]: Done

File: test/e2e/framework/gateways.go
Comment (MaxBab) [reply test/e2e/framework/gateways.go]: Done

File: test/e2e/framework/gateways.go
Comment (tpantelis) [inline test/e2e/framework/gateways.go]: A bit of a nit but this isn't needed.

**Review by tpantelis — CHANGES_REQUESTED** [review]
One minor thing but otherwise looks good.

File: test/e2e/framework/gateways.go
Comment (MaxBab) [reply test/e2e/framework/gateways.go]: Removed.

**Review by tpantelis — APPROVED** [review]

**Comment by MaxBab** [comment]
Hi @skitt @sridhargaddam 

I'll appreciate if you could take a look at this PR.
It already has 1 approval and merging it will unblock me for the next patches.

Thank you!

**Review by skitt — APPROVED** [review]

**Comment by skitt** [comment]
For future reference, your commit and PR titles should give the purpose of the change. In this case, the first line of your commit message would make a good title; something like “Restore gateway node state after tests”.

**Pull request merged**
